lib/classes/Variables.test.js

```diff
@@ -1594,10 +1594,19 @@ module.exports = {
           expect(getValueFromSsmStub).to.have.been.calledWith('ssm:/test/path/to/param');
         }));
 
-    it('should reject invalid sources', () =>
+    it('should reject invalid sources', () => {
       serverless.variables
         .getValueFromSource('weird:source')
-        .should.be.rejectedWith(serverless.classes.Error));
+        .should.be.rejectedWith(serverless.classes.Error);
+
+      serverless.variables
+        .getValueFromSource('weird:"source"')
```

Comment:
I added those tests here because the problem were not with the custom sources, they're working fine and there are tests already for them, the issue was that the stringRef matching was picking customSources values with quotes incorrectly, before the custom sources had a chance to match it. The reason being the resolver attribution which is sequential, following the order where the matchers were added to the array.

In other words, this is what was happening (pseudocode):

```js
const variable = 'echo:"foo bar"';

const resolverList = [
  ...others
  stringMatchRef,
  ...others,
  ...customSourcesResolvers,
]

let ret;

for (let { regex, resolver } of resolverList) {
  if (regex.match(variable)) {
    ret = resolver;
    break;
  }
}

// ret is wrongly set to stringMatchRefResolver, as the variable is not a raw string, but a source named "echo"
ret;

Reply:
Thanks for taking this on @rdsedmundo. I can try to clarify if that is helpful. 

The issue is that whitespace characters do not currently work in variable references.

Take [S3 variable references](https://serverless.com/framework/docs/providers/aws/guide/variables#referencing-s3-objects) as an example. If I try to reference an object `my key` then a reference like `${s3:myBucket/my key}` fails because serverless removes the whitespace and tries to fetch `mykey` instead of `my key`.

This happens with [custom variable types](https://serverless.com/framework/docs/providers/aws/guide/plugins#custom-variable-types) too. The echo plugin example resolves `${echo:foo bar}` to `foobar` instead of `foo bar`

In an effort to try to get whitespace working I tried using single and double quotes but that failed in different ways.

Hope that clarifies the issue.